amethyst_network/src/systems/network_base.rs

```diff
@@ -0,0 +1,37 @@
+//! The network base trait.
+//! Used to share code between the client and the server.
+
+extern crate bincode;
+
+use std::net::UdpSocket;
+use std::clone::Clone;
+use bincode::{serialize, deserialize, Infinite};
+use resources::*;
+
+use serde::Serialize;
+use serde::de::DeserializeOwned;
+use bincode::internal::ErrorKind;
+
+/// The NetworkBase trait is used to share the code between the client and the server network systems.
+/// The T generic parameter corresponds to the network event enum type.
+pub trait NetworkBase<T> where T:Send+Sync+Serialize+Clone+DeserializeOwned+BaseNetEvent<T>+'static{
+
+    /// Sends an event to the target NetConnection using the provided network Socket.
+    /// The socket has to be binded!
+    fn send_event(&self,event:&T,target:&NetConnection,socket:&UdpSocket){
+        let ser = serialize(event, Infinite);
+        match ser{
+            Ok(s)=>{
+                if let Err(e) = socket.send_to(s.as_slice(), target.target){
+                    println!("Failed to send data to network socket: {}",e);
```

Comment:
psst, a logging crate was added while you were away, you can replace this with macros from [`log`](https://docs.rs/log/0.4.1/log/)

Reply:
fixed